tests/test_register.py

```diff
@@ -271,7 +301,75 @@ def test_register_permissions_existing_group_without_intermediate_entries(self):
         path = "/groups/{}/resources/{}/permissions".format(self.test_perm_grp_name, res3_id)
         resp = utils.test_request(self.app, "GET", path)
         body = utils.check_response_basic_info(resp)
-        utils.check_all_equal(body["permission_names"], [res3_perm.value])
+        utils.check_val_is_in(res3_perm, body["permission_names"])
+
+    def test_register_permissions_multiple_resource_type_possible(self):
+        """
+        Verify that service that allows multiple resource-types children retrieves ``type`` field for their creation.
+        """
+        svc_type = ServiceTHREDDS.service_type
+        svc_name = "unittest-service-thredds-register-children-resources"
+        utils.TestSetup.create_TestService(self,
+                                           override_service_name=svc_name,
+                                           override_service_type=svc_type)
+        utils.TestSetup.create_TestGroup(self, override_group_name=self.test_perm_grp_name)
+        session = get_db_session_from_settings(self.app.app.registry.settings)
+
+        res1_name = "test-resource"
+        res2_name = "sub-test-resource"
+        res3_name = "sub-sub-test-resource"
+        res3_perm = "write-deny-recursive"
+        res3_path = res1_name + "/" + res2_name + "/" + res3_name  # none exist, all created for final permission
+        perm_config = {
+            "permissions": [
+                {
+                    "service": svc_name,  # exists
+                    "resource": res3_path,
+                    "permission": res3_perm,  # perm only on lowest child
+                    "type": Directory.resource_type_name,   # without this, fails because cannot infer Directory/File
+                    "action": "create",
+                    "group": self.test_perm_grp_name,
+                },
+            ]
+        }
+        utils.check_no_raise(
+            lambda: register.magpie_register_permissions_from_config(perm_config, db_session=session))
+
+        # check that all service, resources, group are created correctly
+        services = utils.TestSetup.get_RegisteredServicesList(self)
+        utils.check_val_is_in(svc_name, [s["service_name"] for s in services])
+        groups = utils.TestSetup.get_RegisteredGroupsList(self)
+        utils.check_val_is_in(self.test_perm_grp_name, groups)
+        resp = utils.test_request(self.app, "GET", "/services/{}/resources".format(svc_name))
+        body = utils.check_response_basic_info(resp)
+        svc_res = body[svc_name]["resources"]  # type: JSON
+        utils.check_val_is_in(res1_name, [svc_res[r]["resource_name"] for r in svc_res])
+        res1_id = [svc_res[r]["resource_id"] for r in svc_res if svc_res[r]["resource_name"] == res1_name][0]
```

Comment:
This block looks a lot like in test_register_permissions_existing_group_create_some_entries, possibility to factor out some code?